src/main/java/com/cburch/logisim/data/Direction.java

```diff
@@ -79,31 +78,28 @@ public int hashCode() {
   }
 
   public Direction reverse() {
-    if (this == Direction.EAST) return Direction.WEST;
     if (this == Direction.WEST) return Direction.EAST;
     if (this == Direction.NORTH) return Direction.SOUTH;
     if (this == Direction.SOUTH) return Direction.NORTH;
-    return Direction.WEST;
+    return Direction.WEST;    // this == Direction.EAST
   }
 
   public int toDegrees() {
-    if (this == Direction.EAST) return 0;
     if (this == Direction.WEST) return 180;
     if (this == Direction.NORTH) return 90;
     if (this == Direction.SOUTH) return 270;
-    return 0;
+    return 0;     // this == Direction.EAST
   }
 
   public String toDisplayString() {
     return disp.toString();
   }
 
   public double toRadians() {
-    if (this == Direction.EAST) return 0.0;
     if (this == Direction.WEST) return Math.PI;
     if (this == Direction.NORTH) return Math.PI / 2.0;
     if (this == Direction.SOUTH) return -Math.PI / 2.0;
-    return 0.0;
+    return 0.0;     // this == Direction.EAST
```

Comment:
I think the problem in this piece of code is the unfortunate choice of mapping of direction to integer with id. If the numbers went around the directions, (e.g E=0, N=1, W=2, S=3) they could be used to calculate the getLeft, getRight, reverse, toDegrees, and toRadians with simple expressions and sometimes mod 4 rather than all those ifs or cases.

Reply:
Thanks for your analysis and suggestions @davidhutchens! A PR in this direction is welcome. If the choice of mapping is not directly used for serialisation into our `.circ` files, which would require ensuring backward compatibility, I think it would be even worthwhile considering a refactoring of this code.